Failed to build autowiring with /std:c++17 + msvc + windows

Build Autowiring with /std:c++17 with Visual studio 2017(version 15.5.7) on windows failed with following failures, could you help have a look?

D:\Autowiring\src\src\.\autowiring/C++11/filesystem.h(20): error C2874: using-declaration causes a multiple declaration of 'std::experimental::filesystem::v1::path' [D:\Autowiring\build_x86\src\autowiring\test\AutowiringTest.vcxproj]      
D:\Autowiring\src\src\.\autowiring/C++11/filesystem.h(26): error C2874: using-declaration causes a multiple declaration of 'std::experimental::filesystem::v1::filesystem_error' [D:\Autowiring\build_x86\src\autowiring\test\AutowiringTest.vcxproj]

--- a/src/autowiring/C++11/filesystem.h
+++ b/src/autowiring/C++11/filesystem.h
@@ -4,7 +4,7 @@
 //C++17 Filesystem standard
 #if defined(_MSC_VER) && _MSC_VER >= 1900
   #include <filesystem>
-  #if _MSC_VER >= 1914 && __cplusplus >= 201402L
+  #if _MSC_VER >= 1914 && _HAS_CXX17
     // VS 2017 15.7 Preview 2 or newer with /std:c++17
     namespace awfsnamespace = std::filesystem;
   #elif _MSC_VER >= 1910
@@ -18,7 +18,7 @@
 namespace awfsnamespace = autoboost::filesystem;
 #endif
 
-#if !defined(_MSC_VER) || _MSC_VER < 1914 || __cplusplus < 201402L
+#if !defined(_MSC_VER) || _MSC_VER < 1914 || !_HAS_CXX17
 namespace std {
   namespace filesystem {
     using awfsnamespace::path;

--- a/src/autowiring/C++11/filesystem.h
+++ b/src/autowiring/C++11/filesystem.h
@@ -4,7 +4,9 @@
 //C++17 Filesystem standard
 #if defined(_MSC_VER) && _MSC_VER >= 1900
   #include <filesystem>
-#if _MSC_VER >= 1910
+  #if _MSC_VER >= 1914
+    namespace awfsnamespace = std::filesystem;
+  #elif _MSC_VER >= 1910
     namespace awfsnamespace = std::experimental::filesystem;
   #else
     namespace awfsnamespace = std::tr2::sys;
@@ -15,6 +17,7 @@
 namespace awfsnamespace = autoboost::filesystem;
 #endif
 
+#if !defined(_MSC_VER) || _MSC_VER < 1914
 namespace std {
   namespace filesystem {
     using awfsnamespace::path;
@@ -33,3 +36,4 @@ namespace std {
     using awfsnamespace::temp_directory_path;
   }
 }
+#endif

--- a/src/autowiring/C++11/filesystem.h
+++ b/src/autowiring/C++11/filesystem.h
@@ -4,7 +4,7 @@
 //C++17 Filesystem standard
 #if defined(_MSC_VER) && _MSC_VER >= 1900
   #include <filesystem>
-  #if _MSC_VER >= 1914 && _HAS_CXX17
+  #if _MSC_VER >= 1914 && _MSVC_LANG >= 201402L
     // VS 2017 15.7 Preview 2 or newer with /std:c++17
     namespace awfsnamespace = std::filesystem;
   #elif _MSC_VER >= 1910
@@ -18,7 +18,7 @@
 namespace awfsnamespace = autoboost::filesystem;
 #endif
 
-#if !defined(_MSC_VER) || _MSC_VER < 1914 || !_HAS_CXX17
+#if !defined(_MSC_VER) || _MSC_VER < 1914 || _MSVC_LANG < 201402L
 namespace std {
   namespace filesystem {
     using awfsnamespace::path;

--- a/src/autowiring/C++11/filesystem.h
+++ b/src/autowiring/C++11/filesystem.h
@@ -4,7 +4,8 @@
 //C++17 Filesystem standard
 #if defined(_MSC_VER) && _MSC_VER >= 1900
   #include <filesystem>
-  #if _MSC_VER >= 1914
+  #if _MSC_VER >= 1914 && __cplusplus >= 201402L
+    // VS 2017 15.7 Preview 2 or newer with /std:c++17
     namespace awfsnamespace = std::filesystem;
   #elif _MSC_VER >= 1910
     namespace awfsnamespace = std::experimental::filesystem;
@@ -17,7 +18,7 @@
 namespace awfsnamespace = autoboost::filesystem;
 #endif
 
-#if !defined(_MSC_VER) || _MSC_VER < 1914
+#if !defined(_MSC_VER) || _MSC_VER < 1914 || __cplusplus < 201402L
 namespace std {
   namespace filesystem {
     using awfsnamespace::path;

